src/pyvcam_driver.py

```diff
@@ -0,0 +1,537 @@
+        """
+        Returns all settings names paired with their values of a parameter.
+
+        :param param_id: The parameter ID.
+
+        :return: A dictionary containing strings mapped to values.
+        """
+        return self.cam.read_enum(param_id)
+    
+    def binning(self) -> tuple[int, int]:
+        """
+        Returns current serial and parallel binning values
+
+        :return:
+        """
+        return self.cam.binning
+    
+    def set_binning(self, value: Union[tuple[int, int], int]) -> None:
+        """
+        Changes binning. A single value wills et a square binning.
+        Binning cannot be changed directly on the camera; but is used for setting up 
+        acquisitions and returning correctly shaped images returned from get_frame and get_live_frame.
+        Binning settings for individual ROIs is not supported.
+
+        :param value: Desired binning value as a tuple, or int for a square binning.
+
+        :return: None.
+        """
+        self.cam.binning = value
+    
+    def chip_name(self) -> str:
+        """
+        Returns the camera sensor's name as a string.
+
+        :return:
+        """
+        return self.cam.chip_name
+    
+    def post_trigger_delay(self) -> int:
+        """
+        Returns the last acquisition's post-trigger delay reported by the camera in ms.
+        
+        :return:
+        """
+        return self.cam.post_trigger_delay
+    
+    def pre_trigger_delay(self) -> int:
+        """
+        Returns the last acquisition's pre-trigger delay reported by the camera in ms.
+
+        :return:
+        """
+        return self.cam.pre_trigger_delay
+    
+    def scan_line_time(self) -> int:
+        """
+        Returns the scan line time in ns.
+
+        :return:
+        """
+        return self.cam.scan_line_time
+    
+    def sensor_size(self) -> tuple[int, int]:
+        """
+        Returns the sensor size of the current camera in a tuple 
+        in the form (serial sensor size, parallel sensor size)
+
+        :return:
+        """
+        return self.cam.sensor_size
+    
+    def speed_table_index(self) -> int:
+        """
+        Returns the current numerical index of the speed table of the camera.
+        """
+        return self.cam.speed_table_index
+    
+    def set_speed_table_index(self, value: int) -> None:
+        """
+        Changes the current numerical index of the speed table of the camera. 
+        See the Port and Speed Choices section inside the PVCAM User Manual for 
+        a detailed explanation about PVCAM speed tables.
+
+        :param value: Desired speed table index.
+
+        :return: None.
+        """
+        self.cam.speed_table_index = value
+    
+    def temp(self) -> float:
+        """
+        Returns the current temperature of a camera in Celsius.
+
+        :return:
+        """
+        return self.cam.temp
+    
+    def temp_setpoint(self) -> float:
```

Comment:
```suggestion
    def get_temp_setpoint(self) -> float:
```
For symmetry with `get_temp_setpoint`.